pkg/kubelet/cm/devicemanager/plugin/v1beta1/client.go

```diff
@@ -69,8 +69,10 @@ func (c *client) Connect() error {
 		klog.ErrorS(err, "Unable to connect to device plugin client with socket path", "path", c.socket)
 		return err
 	}
+	c.mutex.Lock()
 	c.grpc = conn
 	c.client = client
+	c.mutex.Unlock()
```

Comment:
do Run() and API() need to RLock to access `c.client`?

Reply:
I think the grpc client is thread safe, the race happens with the grpc conn

Reply:
I meant the reassignment to c.client here racing with the access of it. I don't think we need to protect the client methods

Reply:
oh, yeah, the problem here seems that there is a Connect() and Disconnect() in parallel, it does not seem Run() or API() are called at the same. time that Connect(), basically those will not work ... but the fact that Disconnect() already had a mutex makes me think those are the only methods that need it

Reply:
Ok, this looks like a strict improvement then

/lgtm